en/services/parameter_ext.md

```diff
@@ -0,0 +1,238 @@
+
+A component may mitigate this risk by, for example, sending the `PARAM_EXT_ACK` multiple times after a parameter is changed.
+
+
+## Parameter Operations
+
+This section defines the state machine/message sequences for all parameter operations.
+
+
+### Read All Parameters {#read_all}
+
+The read-all operation is started by sending the [PARAM_EXT_REQUEST_LIST](../messages/common.md#PARAM_EXT_REQUEST_LIST) message.
+The target component must start to broadcast the parameters individually in [PARAM_EXT_VALUE](../messages/common.md#PARAM_EXT_VALUE) messages after receiving this message.
+The drone should allow a pause after sending each parameter to ensure that the operation doesn't consume all of the available link bandwidth (30 - 50 percent of the bandwidth is reasonable).
+
+{% mermaid %}
+sequenceDiagram;
+    participant GCS
+    participant Drone
+    GCS->>Drone: PARAM_EXT_REQUEST_LIST
+    GCS-->>GCS: Start receive timeout (any params)
+    Drone->>GCS: Broadcast N parameters with PARAM_EXT_VALUE
+    GCS-->>GCS: Start receive timeout (after each param)
+    Note over GCS: Finish/timeout when no more params received
+{% endmermaid %}
+
+The sequence of operations is:
+
+1. GCS (client) sends [PARAM_EXT_REQUEST_LIST](../messages/common.md#PARAM_REQUEST_LIST) specifying a target system/component.
+   - Broadcast addresses may be used.
+     All targeted components should respond with parameters (or ignore the request if they have none).
+   - The GCS is expected to accumulate parameters from all responding systems.
+   - The timeout/retry behaviour is GSC dependent.
+1. The targeted component(s) should respond, sending all parameters individually in [PARAM_EXT_VALUE](../messages/common.md#PARAM_EXT_VALUE) messages.
+   - Allow breaks between each message in order to avoid saturating the link.
+   - Components with no parameters should ignore the request.
+1. GCS starts timeout after each `PARAM_EXT_VALUE` message in order to detect when parameters are no longer being sent (that the operation has completed).
+
+The GCS/API may accumulate the received parameters for each component and can determine if any are missing/not received (`PARAM_EXT_VALUE` contains the total number of params and index of current param).
+
+**Handling of missing params is GCS-dependent.**
+*QGroundControl*, for example, [individually requests](#read_single) each missing parameter by index (using [PARAM_EXT_REQUEST_READ](../messages/common.md#PARAM_EXT_REQUEST_READ)).
+
+
+### Read Single Parameter {#read_single}
+
+A single parameter can be read by sending the [PARAM_EXT_REQUEST_READ](../messages/common.md#PARAM_EXT_REQUEST_READ) message, as shown below:
+
+{% mermaid %}
+sequenceDiagram;
+    participant GCS
+    participant Drone
+    GCS->>Drone: PARAM_EXT_REQUEST_READ
+    GCS->>GCS: Start receive timeout
+    Drone->>GCS: PARAM_EXT_VALUE
+    GCS-->>Drone: Re-request parameter value on timeout
+{% endmermaid %}
+
```

Comment:
@dogmaphobic IMO it would have been much nicer to get rid of PARAM_EXT_VALUE and always return a PARAM_EXT_ACK. This would allow a read (for example) to let the caller know that it has requested an invalid parameter. Also consistent handling everywhere.